core/domain/feedback_jobs_one_off.py

```diff
@@ -182,3 +184,95 @@ def map(item):
     @staticmethod
     def reduce(key, value):
         yield (key, len(value))
+
+
+class GeneralizeFeedbackThreadMigrationOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """One off job to populate two new fields for instances of Feedback thread
+    model.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [feedback_models.FeedbackThreadModel]
+
+    @staticmethod
+    def map(thread):
+        thread.entity_type = 'exploration'
+        thread.entity_id = thread.exploration_id
+        thread.put()
+
+    @staticmethod
+    def reduce(thread):
+        pass
+
+
+class FeedbackThreadIdMigrationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for migrating instances of feedback thread model and related
+    models to have thread ID of the for entity_type.entity_id.random_str.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [feedback_models.FeedbackThreadModel,
+                feedback_models.FeedbackMessageModel,
+                feedback_models.FeedbackThreadUserModel,
+                email_models.FeedbackEmailReplyToIdModel,
+                user_models.UserSubscriptionsModel]
+
+    @staticmethod
+    def map(item):
+        if isinstance(item, feedback_models.FeedbackThreadModel):
+            old_id = item.id
+            # To make sure the job is idempotent, we create new instances with
+            # the new id. If the id has 2 parts, it needs to be migrated,
+            # else it has already been migrated.
+            if len(old_id.split('.')) == 2:
+                new_id = 'exploration.' + item.id
+                feedback_models.FeedbackThreadModel(
+                    id=new_id, entity_type=item.entity_type,
+                    entity_id=item.entity_id,
+                    exploration_id=item.exploration_id,
+                    state_name=item.state_name,
+                    original_author_id=item.original_author_id,
+                    status=item.status, subject=item.subject,
+                    summary=item.summary, has_suggestion=item.has_suggestion,
+                    message_count=item.message_count,
+                    last_updated=item.last_updated, created_on=item.created_on,
```

Comment:
On looking at BaseModel in core.storage it says that last_updated cannot be set directly. If that's the case then ... you have a problem here, right?

Reply:
Yeah. I went with your solution to override the field in the sub class and then do this (also I added this case to the backend tests and it worked fine after the model change).